chatbot/core/aiml/parse_aiml.php

```diff
@@ -212,7 +212,7 @@ function parse_matched_aiml($convoArr, $type = "normal", $aiml_pattern = '')
     //which debug mode?
     runDebug(__FILE__, __FUNCTION__, __LINE__, "Run the aiml parse in $type mode (normal or srai)", 3);
 
-    $aiml_pattern = ($aiml_pattern == '') ? $convoArr['aiml']['pattern'] : $aiml_pattern;
```

Comment:
We don't necessarily want to change the value of `$aiml_pattern` if it already has a value, but this line of code needs to be written correctly. Probably the following should be implemented, instead?

`$aiml_pattern = (empty($aiml_pattern)) ? $convoArr['aiml']['pattern'] : $aiml_pattern;`